tools/cli/adminTaskListCommands.go

```diff
@@ -128,9 +130,34 @@ func AdminListTaskListTasks(c *cli.Context) {
 	}
 
 	req := &persistence.GetTasksRequest{NamespaceID: namespace, TaskList: tlName, TaskType: tlType, ReadLevel: minReadLvl, MaxReadLevel: &maxReadLvl}
-	tasks, err := taskManager.GetTasks(req)
-	if err != nil {
-		ErrorAndExit("Failed to get Tasks", err)
+	paginationFunc := func(paginationToken []byte) ([]interface{}, []byte, error) {
+		response, err := taskManager.GetTasks(req)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		tasks := response.Tasks
+		if workflowID != "" {
+			all := tasks
```

Comment:
You might want to use this: https://github.com/golang/go/wiki/SliceTricks#filter-in-place (read entire page, it is fun and useful).

Reply:
or this: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating